operator/pkg/controllers/configconnectorcontext/configconnectorcontext_controller_test.go

```diff
@@ -896,6 +897,13 @@ func TestHandleReconcileFailed(t *testing.T) {
 	if err := c.Create(ctx, ccc); err != nil {
 		t.Fatalf("failed to create ConfigConnectorContext: %v", err)
 	}
+
+	// Manually set the observed generation, simulating the declarative reconciler.
+	ccc.Status.ObservedGeneration = 1
```

Comment:
From my reading of the code it seems like the configconnectorcontext controller implements its own v1alpha1 CommonStatus builder as opposed to using one of the [k-d-p status builders](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/tree/master/pkg/patterns/addon/pkg/status), so I expect the controller will need to be updated to set the observedGeneration. As is it doesn't look like this unit test actually validates that the controller sets the observedGeneration, just that the controller doesn't clobber the observedGeneration